service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/EhrExtractMapper.java

```diff
@@ -0,0 +1,111 @@
+package uk.nhs.adaptors.gp2gp.ehr;
+
+import java.util.Optional;
+
+import uk.nhs.adaptors.gp2gp.common.service.FhirParseService;
+import uk.nhs.adaptors.gp2gp.common.service.XPathService;
+import uk.nhs.adaptors.gp2gp.common.task.TaskIdService;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrFhirExtractParams;
+import uk.nhs.adaptors.gp2gp.ehr.utils.TemplateUtils;
+
+import org.hl7.fhir.dstu3.model.Bundle;
+import org.hl7.fhir.dstu3.model.Patient;
+import org.hl7.fhir.dstu3.model.Reference;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import org.w3c.dom.Document;
+import org.xml.sax.SAXException;
+
+import com.github.mustachejava.Mustache;
+
+@Component
+public class EhrExtractMapper {
+    private static final Mustache EHR_EXTRACT_TEMPLATE = TemplateUtils.loadTemplate("ehr_extract_template.mustache");
+    private static final String EHR_REQUEST_PATH = "/RCMR_IN010000UK05/ControlActEvent/subject/EhrRequest";
+    private static final String EHR_REQUEST_ID_PATH = EHR_REQUEST_PATH + "/id/@root";
+    private static final String EHR_REQUEST_PATIENT_ID_PATH = EHR_REQUEST_PATH + "/recordTarget/patient/id";
+    private static final String EHR_REQUEST_PATIENT_ID_EXTENSION_PATH = EHR_REQUEST_PATIENT_ID_PATH + "/@extension";
+    private static final String EHR_AUTHOR_ID_PATH = EHR_REQUEST_PATH + "/author/AgentOrgSDS/agentOrganizationSDS/id";
+    private static final String EHR_AUTHOR_ID_EXTENSION_PATH = EHR_AUTHOR_ID_PATH + "/@extension";
+    private static final String EHR_DESTINATION_ID_PATH = EHR_REQUEST_PATH + "/destination/AgentOrgSDS/agentOrganizationSDS/id";
+    private static final String EHR_DESTINATION_ID_EXTENSION_PATH = EHR_DESTINATION_ID_PATH + "/@extension";
+    private static final String SLASH = "/";
+
+    private final XPathService xPathService;
+    private final FhirParseService fhirParseService;
+    private final TaskIdService taskIdService;
+
+    @Autowired
+    public EhrExtractMapper(XPathService xPathService, FhirParseService fhirParseService, TaskIdService taskIdService) {
+        this.xPathService = xPathService;
+        this.fhirParseService = fhirParseService;
+        this.taskIdService = taskIdService;
+    }
+
+    public EhrFhirExtractParams mapXmlToEhrFhirExtractParams(String xml) {
```

Comment:
Doesn't EhrRequestHandler already extract most of if not all of these values?

We'll want anything needed from the original EHR Extract message to be included in the `GetGpcStructuredTaskDefinition`.

When we do the translation the original request XML will not be available so we cannot get those values in this way.